geoh5py/workspace/workspace.py

```diff
@@ -1236,6 +1246,13 @@ def open(self, mode: str | None = None) -> Workspace:
         if mode is None:
             mode = self._mode
 
+        if mode == "r+" and any(k in str(self.h5file) for k in NETWORK_DRIVES):
```

Comment:
```suggestion
        if mode == "r+" and any(k in str(self.h5file.absolute()) for k in NETWORK_DRIVES):
```
Need to also make sure h5file is a `str` of `Path`.